deploy/roles/spark/tasks/main.yml

```diff
@@ -0,0 +1,86 @@
+---
+- name: Create service account for Spark
+  user: name={{ spark_user }}
+        system=yes
+        home={{ spark_lib_dir }}
+        shell={{ spark_user_shell }}
+        state=present
+        groups="{{ spark_user_groups | join(',') }}"
+  tags: ["spark-user"]
```

Comment:
How are you using tags in this case?  Before I've set a tag name based on the role name, to make it easy to run a single role when doing development/test, but it seems you have a more complex scheme here.  This might be a good idea, just curious what your plan is.